### DISCUSSION TITLE
Pass the text to format directly as an argument?

### DISCUSSION THREAD

**Discussion by hbina** [prettier/prettier#14353] [Q&A] [answered]
Is it possible to support this kind of use case? I have a plugin (https://github.com/hbina/lapce-prettier) for Lapce that currently only supports taking in args. I would like to use this feature if possible instead of piping in the text which is cumbersome.

Something like `prettier --text "hello world"`. It should handle text escape and all that of course.

**Comment by tamish-max** [comment]
There isn't a `--text` flag. The normal CLI way is stdin + `--stdin-filepath` so Prettier can infer the parser:

```bash
echo "const x=1" | prettier --stdin-filepath example.js
```

`--stdin-filepath` matters because parser inference is based on the filename. If you're building an editor/plugin, using the Node API (`prettier.format(...)`) is usually cleaner than shelling out.

**Comment by fisker** [comment]
```bash
echo "code" | prettier --parser=babel
```

**Comment by Poliklot** [comment, marked as answer]
@fisker’s answer is the right one for Prettier CLI:

```bash
echo "code" | prettier --parser=babel
```

or, even better for editor integrations:

```bash
prettier --stdin-filepath file.js
```

with the document text passed through stdin.

I checked the Lapce plugin linked in the original post, and the limitation seems to be on the integration side, not on Prettier’s side.

At the moment, that plugin gets the file path from `text_document.uri` and runs Prettier like this:

```rust
vec![
    "--write".to_string(),
    file_path,
]
```

Source: https://github.com/hbina/lapce-prettier/blob/master/src/main.rs#L44-L68

So it is effectively doing:

```bash
prettier --write file.js
```

There is also a comment in the plugin:

```rust
// Hack because we can't send the document in the result
// Remove this once `execute_process` supports stdout/stderr
```

Source: https://github.com/hbina/lapce-prettier/blob/master/src/main.rs#L59-L65

So Prettier already supports formatting text via stdin. The missing part is that this Lapce plugin would need a way to send the current buffer to Prettier’s stdin and read formatted output from stdout, or use a temp file / another bridge if the plugin API still does not support that cleanly.

So I don’t think Prettier needs a separate `--text` option here. This looks more like something to fix in the editor/plugin integration layer.
